BHBC-1925

cgarrettjones opened the pull request:
# Overview

- adjusted permit model

## Links to Jira tickets

- [BHBC-1925](https://quartech.atlassian.net/browse/BHBC-1925)

## Description of relevant changes

- adjusted permit model for model 1.2
- adjusted db function project deletion routines
- added smoke test file

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

cgarrettjones committed:
- adjusted permit model for model 1.2
- adjusted db function project deletion routines
- added smoke test file

curtisupshall reviewed (dismissed):
Looks good to my eyes. 👍

AlfredRosenthal commented:
This all looks good. I did try running it locally just to see what happens and I think the removal of `project_id` from the `permit` table means we'll have to update a few SQL statements that are still expecting that to exist. That might have been the blow up you were seeing when you were trying to run things @cgarrettjones.

NickPhura committed:
Purge permit code from everywhere except anything under survey files.
- App/Api loads without error.
- Projects can be created/displayed.
- TODO: Creating a survey is still broken.

NickPhura committed:
Add CustomFieldArray Component

NickPhura committed:
Bugfix for previous commit

cgarrettjones committed:
- merging work associated with BHBC-1940
- added user and survey based permit listing

cgarrettjones committed:
- minor api path change

cgarrettjones committed:
- test file name change

cgarrettjones committed:
- adjustment of permit api from noun to verb based per current app convention

anissa-agahchen committed:
commenting out required permit

anissa-agahchen committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

anissa-agahchen committed:
basic create in front-end is working

cgarrettjones committed:
- data model changes to remove unused permit attributes

cgarrettjones committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

anissa-agahchen committed:
merge conflicts

cgarrettjones committed:
- completion of api permits CRUD

cgarrettjones committed:
- minor log label corrections

cgarrettjones committed:
- removed concept of permit first class object

anissa-agahchen committed:
createSurveyPage

anissa-agahchen committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

NickPhura committed:
Wire up survey create endpoint to permit service

NickPhura committed:
Wire up survey update and view to permit service

NickPhura committed:
Survey permit updates

anissa-agahchen committed:
yup schema updates

anissa-agahchen committed:
updated snapshots

anissa-agahchen committed:
fix warnings

anissa-agahchen committed:
tests for SurveyPermitForm

NickPhura committed:
WIP updates to API tests

NickPhura committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

anissa-agahchen committed:
updated tests

anissa-agahchen committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

NickPhura committed:
Remove unused paths

NickPhura committed:
Misc cleanup

NickPhura committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

NickPhura committed:
Update tests for permit/list

anissa-agahchen committed:
GetPermitData test

anissa-agahchen committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into BHBC-1925

NickPhura committed:
Fix app formatting/linting

AlfredRosenthal commented on api/src/services/eml-service.ts:
```diff
@@ -409,37 +409,39 @@ export class EmlService extends DBService {
-        }
-      });
-    }
+    // TODO add back when survey supports permits
```
should these be added back now?

NickPhura replied:
we may never add them back tbh, as its questionable if permits belong in the EML at all.  But not certain yet.

AlfredRosenthal commented on api/src/services/permit-service.test.ts:
```diff
@@ -1,313 +1,10 @@
-
-      expect(res).to.equal(12);
-    });
+  it('TODO', () => {
```
could be removed

NickPhura replied:
Removed, and added real tests (was just a placeholder so as not to forget)

jeznorth committed:
UI Fixes

NickPhura committed:
Api test updates

NickPhura committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

NickPhura committed:
Remove commented code (code smell)

AlfredRosenthal commented on api/src/services/survey-service.ts:
```diff
@@ -491,31 +489,52 @@ export class SurveyService extends DBService {
+    // Get existing permits
+    const existingPermits = await permitService.getPermitBySurveyId(surveyId);
+
+    // Determine if any existing permits need to be deleted (are not included in the incoming permits array)
```
comment is a little confusing maybe change to
'Incoming permits array is our source of truth, find and remove any hanging/ leftover permits'

NickPhura replied:
Took a stab at updating all of these comments to try and be more descriptive, not sure if its better or just different :)

AlfredRosenthal replied:
I think it's better!

cgarrettjones committed:
- adjustments to permit service to supply user id

cgarrettjones committed:
Merge branch 'BHBC-1925' of https://github.com/bcgov/biohubbc into BHBC-1925

NickPhura committed:
Update frontend tests

NickPhura committed:
Updates tests

NickPhura committed:
Fix test

AlfredRosenthal reviewed (approved):
🐶
